lzy/execution-env/src/main/java/ai/lzy/env/aux/CondaPackageRegistry.java

```diff
@@ -14,167 +21,297 @@ public class CondaPackageRegistry {
+
+        var name = (String) res.getOrDefault("name", "default");
+
+        Map<String, Package> pkgs = new HashMap<>();
+
+        //noinspection rawtypes
+        var deps = (List) res.getOrDefault("dependencies", List.of());
+
+        String pythonVersion = null;
+        String pypiIndex = null;
+
+        for (var dep : deps) {
+            if (dep instanceof String) {
+                var dat = ((String) dep).split(VERSION_REGEX, SPLIT_LIMIT);
+                if (dat[0].equals("python")) {
+                    pythonVersion = dat.length > 1 ? dat[1] : null;
+                }
+
+                //skipping other conda dependencies
+            }
+
+            //noinspection rawtypes
+            if (dep instanceof Map depsMap) {
+                var pipDeps = depsMap.get("pip");
+                if (pipDeps == null) {
+                    return null;
+                }
+                if (!(pipDeps instanceof List)) {
+                    return null;
                 }
 
-                //noinspection rawtypes
-                if (dep instanceof Map depsMap) {
-                    var pipDeps = depsMap.get("pip");
-                    if (pipDeps == null) {
-                        return;
+                //noinspection unchecked
+                for (var pipDep : (List<Object>) pipDeps) {
+                    if (!(pipDep instanceof String)) {
+                        return null;
                     }
-                    if (!(pipDeps instanceof List)) {
-                        return;
+
+                    if (((String) pipDep).startsWith("--index-url")) {
+                        var parts = ((String) pipDep).split(" ");
+                        pypiIndex = parts.length > 1 ? parts[1] : null;
+                        continue;
                     }
 
-                    //noinspection unchecked
-                    for (var pipDep : (List<Object>) pipDeps) {
-                        if (!(pipDep instanceof String)) {
-                            return;
-                        }
-                        var dat = ((String) pipDep).split(VERSION_REGEX, SPLIT_LIMIT);
-                        if (dat.length == 1) {
-                            pkgs.put(dat[0], new Package(dat[0], "some-very-strange-version"));
-                        } else {
-                            pkgs.put(dat[0], new Package(dat[0], dat[1]));
-                        }
+                    var dat = ((String) pipDep).split(VERSION_REGEX, SPLIT_LIMIT);
+
+                    var pkgName = normalizePkgName(dat[0]);
+                    if (dat.length == 1) {
+                        pkgs.put(pkgName, new Package(pkgName, null));
+                    } else {
+                        pkgs.put(pkgName, new Package(pkgName, dat[1]));
                     }
                 }
             }
+        }
 
-            envs.put(name, pkgs);
+        if (pythonVersion != null) {
+            var splitPythonVersion = pythonVersion.split("\\.", 3);
```

Comment:
what about `rc` versions?
`1.1.0rc5` for example

Reply:
`rc` python versions?
In this case we ignoring patch number, so we will install `python=1.1`